### PR TITLE
Feature/apps 1790 migration to gha

### PR DESCRIPTION
**Pull request by omkaryadv177**
Migration of travis CI to Github action

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+          aws-region: eu-west-1
+
+      - name: Run Script
+        run: |
+          bash _ci/init.sh
+          source _ci/prepare_staging_deploy.sh
+
+      - name: "Release to S3 Staging Bucket"
+        run: | 
+           aws s3 cp --recursive ./deploy_dir_community s3://alfresco-artefacts-staging/enterprise/GoogleDocs/${VERSION}
```
Comment (kavitshah-gl) [inline .github/workflows/build.yml:118]: Shouldn't it be s3://alfresco-artefacts-staging/community/GoogleDocs/${VERSION}

**Review by kavitshah-gl — CHANGES_REQUESTED** [review]

File: _ci/release.sh [hunk 2]
```diff
@@ -6,17 +6,17 @@ set -vex
 pushd "$(dirname "${BASH_SOURCE[0]}")/../"
 
 # For PR builds only execute a Dry Run of the release
-[ "${TRAVIS_PULL_REQUEST}" = "false" ] && DRY_RUN="" || DRY_RUN="-DdryRun"
+ DRY_RUN="" || DRY_RUN="-DdryRun"
 
 # Travis CI runner work on DETACHED HEAD, so we need to checkout the release branch
-git checkout -B "${TRAVIS_BRANCH}"
+git checkout -B "$GITHUB_RUN_NUMBER"
```
Comment (kavitshah-gl) [inline _ci/release.sh:12]: This should be branch instead of Run Number

**Review by kavitshah-gl — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      - name: Run ${{ matrix.stage }}
+        if: github.ref_name != 'company_release'
+        run: |
+           bash _ci/init.sh
+           bash _ci/${{ matrix.stage }}.sh
+
+  Release:
+    if: ${{ ( startsWith(github.ref_name, 'HF/')  || startsWith(github.ref_name, 'SP/') || github.ref_name == 'master' )  &&
+        contains(github.event.head_commit.message, '[trigger release]') || inputs.commitMessage == '[trigger release]'  &&
+        github.event_name != 'pull_request' }}
```
Comment (kavitshah-gl) [inline .github/workflows/build.yml:70]: In the Travis.yml we don't have this condition to check if event in not pull request.

File: _ci/release.sh [hunk 4]
```diff
@@ -6,17 +6,17 @@ set -vex
 pushd "$(dirname "${BASH_SOURCE[0]}")/../"
 
 # For PR builds only execute a Dry Run of the release
-[ "${TRAVIS_PULL_REQUEST}" = "false" ] && DRY_RUN="" || DRY_RUN="-DdryRun"
+[ "$GITHUB_EVENT" != "pull_request" ] && DRY_RUN="" || DRY_RUN="-DdryRun"
```
Comment (kavitshah-gl) [inline _ci/release.sh:9]: This condition should be similar to Alfresco Extension Inspector.

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+      - name: Run Script Release to nexus
+        run: |
+          bash _ci/init.sh
+          bash _ci/release.sh
+          source _ci/prepare_staging_deploy.sh
+
+      - name: "Release to S3 Staging Bucket"
+        run: | 
+           aws s3 cp --recursive ./deploy_dir_community s3://alfresco-artefacts-staging/community/GoogleDocs/${VERSION}
```
Comment (kavitshah-gl) [inline .github/workflows/build.yml:114]: How is the value of version going to come?

**Review by kavitshah-gl — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 5 above)
Comment (omkaryadv177) [reply .github/workflows/build.yml:114]: it is exporting the value of version from prepare_staging_deploy.sh script file

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+
+      - name: "Configure AWS credentials"
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_S3_RELEASE_ACCESS_KEY_ID }}
```
Comment (kavitshah-gl) [inline .github/workflows/build.yml:101]: For staging you should use staging access keys for AWS.

**Review by kavitshah-gl — CHANGES_REQUESTED** [review]

**Review by kavitshah-gl — APPROVED** [review]
Looks good to me with all the changes.

**Pull request merged**